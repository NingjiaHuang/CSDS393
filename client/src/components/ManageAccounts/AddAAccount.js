import React, {useState} from 'react'
import { Button, Modal } from 'react-bootstrap';

const AddAAccount = () =>{
    const [info, setInfo] = useState({
        username:"",
        user_password:"",
        account_type:"admini",
        reg_email:"",
        reg_phone:"",
    });

    const {username, user_password,account_type, reg_email, reg_phone} = info;
        
    const onChange = (e) => {
        setInfo({...info, [e.target.name]:e.target.value})
    }

    const [show, setShow] = useState(false);
    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);

    const onSubmitForm = async(e) => {
        e.preventDefault();
        try{
            const body = {username, user_password,account_type, reg_email, reg_phone};
            const response = await fetch("http://localhost:4020/api/v1/cats",{ // url needs update. need to add to breeding cat table
                method:"Post",
                headers: {"Content-Type": "application/json"},
                body: JSON.stringify(body)
            });

            window.location="/";
        } catch(err){
            console.error(err.message);
        }
    }



    return(
        <div>
                <Button data-testid="showBtn" variant="info" onClick={handleShow}>Add an admin account</Button>
                <Modal show={show} onHide={handleClose}>
                    <Modal.Header closeButton>
                    <Modal.Title>Add a breeding cat</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>
                        <form onSubmit={onSubmitForm}>
                            <label>Username: </label>
                            <input type="text" 
                            name="username"    
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label>Password: </label>
                            <input 
                            type="text" 
                            name="user_password"  
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label>Account type: </label>
                            <input 
                            placeholder = "admini"
                            type="text" 
                            name="account_type"  
                            className="form-control my-2"/>

                            <label>Email: </label>
                            <input 
                            type="text" 
                            name="reg_email"   
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label>Phone: </label>
                            <input type="text" 
                            name="reg_phone"   
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>
                        </form>
                    </Modal.Body>
                    <Modal.Footer>
                    <Button variant="secondary" onClick={handleClose}>
                        Close
                    </Button>
                    <Button variant="warning" onClick={onSubmitForm, handleClose}>
                        Add
                    </Button>
                    </Modal.Footer>
                </Modal>
        </div>
    )
}

export default AddAAccount;