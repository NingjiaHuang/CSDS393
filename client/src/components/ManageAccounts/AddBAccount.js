import React, {useState} from 'react'
import { Button, Modal } from 'react-bootstrap';

const AddBAccount = () =>{
    const [info, setInfo] = useState({
        username:"",
        user_password:"",
        account_type:"breeder",
        reg_email:"",
        reg_phone:"",
        cattery_name:"",
        organization:"",
        owner_name:"",
        city:""
    });

    const {username, user_password,account_type, reg_email, reg_phone, 
        cattery_name, organization, owner_name, city} = info;
    
    const onChange = (e) => {
        setInfo({...info, [e.target.name]:e.target.value})
    }
    
    const [show, setShow] = useState(false);
    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);
    const onSubmitForm = async(e) => {
        e.preventDefault();
        try{
            const body = {username, user_password,account_type, reg_email, reg_phone, 
                cattery_name, organization, owner_name, city};
            const response = await fetch("http://localhost:4020/api/v1/cats",{
                method:"Post",
                headers: {"Content-Type": "application/json"},
                body: JSON.stringify(body)
            });

            window.location="/";
        } catch(err){
            console.error(err.message);
        }
    }



    return(
        <div>
                <Button data-testid="showBtn" variant="info" onClick={handleShow}>Add breeder account</Button>
                <Modal show={show} onHide={handleClose}>
                    <Modal.Header closeButton>
                    <Modal.Title>Add a kitten</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>
                    <form onSubmit={onSubmitForm}>
                            <label>Username: </label>
                            <input type="text" 
                            name="username"    
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label>Password: </label>
                            <input 
                            type="text" 
                            name="user_password"  
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label>Account type: </label>
                            <input 
                            placeholder = "breeder"
                            type="text" 
                            name="account_type"  
                            className="form-control my-2"/>

                            <label>Email: </label>
                            <input 
                            type="text" 
                            name="reg_email"   
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label>Phone: </label>
                            <input type="text" 
                            name="reg_phone"   
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label>Cattery name: </label>
                            <input type="text" 
                            name="cattery_name"   
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label>Organization: </label>
                            <input type="text" 
                            name="organization"   
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label>Owner name: </label>
                            <input type="text" 
                            name="owner_name"   
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label>City: </label>
                            <input type="text" 
                            name="city"   
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>
                        </form>
                    </Modal.Body>
                    <Modal.Footer>
                    <Button variant="secondary" onClick={handleClose}>
                        Close
                    </Button>
                    <Button variant="warning" onClick={onSubmitForm, handleClose}>
                        Add
                    </Button>
                    </Modal.Footer>
                </Modal>
        </div>
    )
}

export default AddBAccount;