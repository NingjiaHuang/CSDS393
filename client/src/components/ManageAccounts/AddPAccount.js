import React, { useState} from 'react'
import { Button, Modal} from 'react-bootstrap';

const AddPAccount = () =>{
    const [info, setInfo] = useState({
        username:"",
        user_password:"",
        account_type:"potential_parents",
        reg_email:"",
        reg_phone:"",
        preferred_name:""
    });

    const {username, user_password,account_type, reg_email, reg_phone, preferred_name} = info;
      
    const onChange = (e) => {
        setInfo({...info, [e.target.name]:e.target.value})
    }

    const [show, setShow] = useState(false);
    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);

    const onSubmitForm = async(e) => {
        e.preventDefault();
        try{
            const body = {username, user_password,account_type, reg_email, reg_phone, preferred_name};
            const response = await fetch("http://localhost:4020/api/v1/cats",{
                method:"Post",
                headers: {"Content-Type": "application/json"},
                body: JSON.stringify(body)
            });

            window.location="/";
        } catch(err){
            console.error(err.message);
        }
    }



    return(
        <div>
                <Button data-testid="showBtn" variant="info" onClick={handleShow}>Add a potential parent account</Button>
                <Modal show={show} onHide={handleClose}>
                    <Modal.Header closeButton>
                    <Modal.Title>Add a pregnant cat</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>
                    <form onSubmit={onSubmitForm}>
                            <label>Username: </label>
                            <input type="text" 
                            name="username"    
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label>Password: </label>
                            <input 
                            type="text" 
                            name="user_password"  
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>


                            <label>Email: </label>
                            <input 
                            type="text" 
                            name="reg_email"   
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label>Phone: </label>
                            <input type="text" 
                            name="reg_phone"   
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>

                            <label>Preferred name: </label>
                            <input type="text" 
                            name="preferred_name"   
                            className="form-control my-2"
                            onChange={e => onChange(e)}/>
                        </form>
                    </Modal.Body>
                    <Modal.Footer>
                    <Button variant="secondary" onClick={handleClose}>
                        Close
                    </Button>
                    <Button variant="warning" onClick={onSubmitForm, handleClose}>
                        Add
                    </Button>
                    </Modal.Footer>
                </Modal>
        </div>
    )
}

export default AddPAccount;
