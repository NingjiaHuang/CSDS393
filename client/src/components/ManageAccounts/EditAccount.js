import React,{Fragment, useState} from 'react';
import {Modal, Button} from 'react-bootstrap';


const EditAccount = ({account}) =>{
    const [show, setShow] = useState(false);
    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);
    const [inputs, setInputs] = useState({
        username: account.username,
        user_password: account.user_password,
        account_type:  account.account_type,
        reg_email: account.reg_email,
        reg_phone: account.reg_phone,
    }); 
    const {username, user_password, account_type, reg_email, reg_phone} = inputs;
    const onChange = (e) => {
        setInputs({...inputs, [e.target.name]:e.target.value})
    }


    //update cat info function
    const updateAccountInfo = async(e) =>{
        e.preventDefault();
        try{
            const body = {username, user_password, account_type, reg_email, reg_phone};
            console.log(body)
            const response = await fetch("http://localhost:4020/auth/update_account",{
                method:"PATCH",
                headers: {"Content-type": "application/json"},
                body: JSON.stringify(body)
            });

            window.location = "/manage_user_account";
        }catch(err){
            console.error(err.message)
        }
    }


    return(
        <Fragment>

            <Button type="button" 
            class="btn btn-warning" 
            data-testid="showbtn"
            onClick={handleShow}>
            Edit
            </Button>
            
            <Modal show={show} onHide={handleClose}>
                <Modal.Header closeButton>
                <Modal.Title>Edit Account</Modal.Title>
                </Modal.Header>
                <Modal.Body>
                    <form onSubmit={updateAccountInfo}>
                        <label htmlFor="username">Username: </label>
                        <input type="text" 
                        name="username"    
                        id="username"
                        placeholder= {account.username}
                        onChange={e => onChange(e)}
                        className="form-control my-2"/>

                        <label htmlFor="password">Password: </label>
                        <input 
                        type="text" 
                        id="password"
                        name="password"  
                        placeholder={account.user_password}  
                        onChange={e => onChange(e)}
                        className="form-control my-2"/>

                        <label  htmlFor="account_type">Account Type: </label>
                        <input 
                        type="text" 
                        id="account_type"
                        name="account_type"  
                        placeholder={account.account_type}  
                        onChange={e => onChange(e)}
                        className="form-control my-2"/>

                        <label htmlFor="reg_phone">Phone: </label>
                        <input type="text" 
                        name="reg_phone"   
                        id="reg_phone"
                        placeholder={account.reg_phone} 
                        onChange={e => onChange(e)}
                        className="form-control my-2"/>
                    </form>
                </Modal.Body>
                <Modal.Footer>
                    <Button data-testid ="closebtn" variant="secondary" onClick={handleClose}>
                        Close
                    </Button>
                    <Button variant="primary" onClick={updateAccountInfo}>
                        Save Changes
                    </Button>
                </Modal.Footer>

                </Modal>
        </Fragment>
    )
}

export default EditAccount;
