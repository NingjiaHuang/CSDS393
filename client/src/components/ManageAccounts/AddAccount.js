import React, {Fragment, useState} from 'react'
import { Button } from 'react-bootstrap';
import AddAAccount from "./AddAAccount"
import AddBAccount from "./AddBAccount"
import AddPAccount from "./AddPAccount"

const AddAccount = () =>{
    return(
        <div className = 'mx-auto'>
             <table class="mx-auto" size="sm">
                 <thead>
                    <th><AddBAccount/></th>
                    <th><AddAAccount/></th>
                    <th><AddPAccount/></th>
                 </thead>
             </table>
        </div>
    )
}

export default AddAccount;