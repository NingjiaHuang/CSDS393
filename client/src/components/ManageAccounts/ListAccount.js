import React, {Fragment, useEffect, useState} from 'react';
import mock_data from "../MOCK_DATA.json"
import EditAccount from "./EditAccount"

const ListAccount = () =>{
    const[searchTerm, setSearchTerm] = useState("")
    const[accounts, setAccounts] = useState([]);

    //delete cat function
    const deleteAccount = async(reg_email) =>{
        try{
            const deleteAccount = await fetch("http://localhost:4020/auth/delete",{
                method: "DELETE"
            });
            setAccounts(accounts.filter(account => account.reg_email !== reg_email))
        }catch (err){
            console.error(err.message);
        }
    }

    const getAccounts = async() => {
        try{
            
            const response = await fetch("http://localhost:4020/auth/all_accounts")
            const jsonData = await response.json()
            setAccounts(jsonData);

        } catch (err){
            console.error(err.message);
        }
    }

    useEffect(() =>{
        getAccounts();
    }, [])

    return (
    <Fragment>
        {" "}
        <input type="text" className="form-control" 
          placeholder="search cats by name here" 
          data-testid="searchAccount"
          onChange={event => {setSearchTerm(event.target.value)}}
          />
        <table class="table mt-5 text-center">
            <thead>
                <tr>
                    <th>Username</th>
                    <th>Account type</th>
                    <th>Email</th>
                    <th>Edit</th>
                    <th>Delete</th>
                </tr>
            </thead>
            <tbody>
            {accounts.filter((val) =>{  //if get all catteries completed, change "mock_data" to catteries
                  if (searchTerm == ""){
                    return val
                  } else if (val.username.toLowerCase().includes(searchTerm.toLowerCase()) ){
                    return val
                  }
              }
              ).map((val, key) => (
                  <tr key={key}>
                      <td>{val.username}</td>
                      <td>{val.account_type}</td>
                      <td>{val.reg_email}</td>
                        <td><EditAccount  cat = {val}/></td>
                    <td>
                        <button className="btn btn-danger"
                        onClick={() => deleteAccount(val.reg_email)}>Delete</button>
                        </td>
                </tr>
            ))
            }
            </tbody>
        </table>
    </Fragment>
    )
}
export default ListAccount;