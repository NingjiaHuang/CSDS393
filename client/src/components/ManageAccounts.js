import React from 'react'
import AddAccount from './ManageAccounts/AddAAccount'
import ListAccount from './ManageAccounts/ListAccount'
import { MDBContainer } from 'mdbreact';

export default function ManageAccounts() {
    return (
        <MDBContainer>
        <div className='mx-auto mt-5'>
          <h2 className="text-center">Manage system accounts</h2>
          <AddAccount />
          <div className='mt-3'>
          <ListAccount />
          </div>
        </div>
      </MDBContainer>
    )
}