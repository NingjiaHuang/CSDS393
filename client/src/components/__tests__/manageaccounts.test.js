import {render, screen, fireEvent, cleanup} from '@testing-library/react';
import AddAAccount from '../ManageAccounts/AddAAccount';
import EditAccount from '../ManageAccounts/EditAccount';
import ListAccount from '../ManageAccounts/ListAccount';
import ManageAccounts from '../ManageAccounts';
import AddAccount from '../ManageAccounts/AddAccount'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect';
afterEach(() =>{
    cleanup();
})

test('matches snapshot of add an admin aacount', () =>{
    const tree = renderer.create(<AddAAccount/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('matches snapshot of add account', () =>{
    const tree = renderer.create(<AddAccount/>).toJSON();
    expect(tree).toMatchSnapshot();
})


test('matches snapshot of list account', () =>{
    const tree = renderer.create(<ListAccount/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('matches snapshot of manage account', () =>{
    const tree = renderer.create(<ManageAccounts/>).toJSON();
    expect(tree).toMatchSnapshot();
})