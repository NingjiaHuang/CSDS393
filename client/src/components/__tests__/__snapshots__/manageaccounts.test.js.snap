// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of add account 1`] = `
<div
  className="mx-auto"
>
  <table
    class="mx-auto"
    size="sm"
  >
    <thead>
      <th>
        <div>
          <button
            className="btn btn-info"
            data-testid="showBtn"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Add an admin account
          </button>
        </div>
      </th>
    </thead>
  </table>
</div>
`;

exports[`matches snapshot of add an admin aacount 1`] = `
<div>
  <button
    className="btn btn-info"
    data-testid="showBtn"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Add an admin account
  </button>
</div>
`;

exports[`matches snapshot of list account 1`] = `
Array [
  " ",
  <input
    className="form-control"
    data-testid="searchAccount"
    onChange={[Function]}
    placeholder="search accounts by username here"
    type="text"
  />,
  <table
    class="table mt-5 text-center"
  >
    <thead>
      <tr>
        <th>
          Username
        </th>
        <th>
          Account type
        </th>
        <th>
          Email
        </th>
        <th>
          Edit
        </th>
        <th>
          Delete
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;

exports[`matches snapshot of manage account 1`] = `
<div
  className="container"
  data-test="container"
>
  <div
    className="mx-auto mt-5"
  >
    <h2
      className="text-center"
    >
      Manage system accounts
    </h2>
    <div
      className="mx-auto"
    >
      <table
        class="mx-auto"
        size="sm"
      >
        <thead>
          <th>
            <div>
              <button
                className="btn btn-info"
                data-testid="showBtn"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Add an admin account
              </button>
            </div>
          </th>
        </thead>
      </table>
    </div>
    <div
      className="mt-3"
    >
       
      <input
        className="form-control"
        data-testid="searchAccount"
        onChange={[Function]}
        placeholder="search accounts by username here"
        type="text"
      />
      <table
        class="table mt-5 text-center"
      >
        <thead>
          <tr>
            <th>
              Username
            </th>
            <th>
              Account type
            </th>
            <th>
              Email
            </th>
            <th>
              Edit
            </th>
            <th>
              Delete
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
